lib/validateCustomDirection.js

```diff
@@ -0,0 +1,23 @@
+module.exports = function () {
+    if (this.direction === 'custom') {
+        if (!areDefinedAndNumbers([
+            this.customDirection.x0,
+            this.customDirection.x1,
+            this.customDirection.y0,
+            this.customDirection.y1
+        ])) {
+            throw new Error('When using a custom direction, the custom object is required with it\'s attributes: x0, x1, y0, y1 of type number');
+        }
+    }
+}
+
+function areDefinedAndNumbers(array) {
+    var definedAndNumber = true, i = 0;
+    while (definedAndNumber && i < array.length) {
+        if (typeof array[i] !== 'number') {
```

Comment:
Don't check if it's a number because the input should be a percentage.
Check if the user input doesn't ends by the character `'%'` and `!Number.isInteger(parseInt(array[i].split('%')[0], 10))`

Reply:
Maybe we could also let the use put pixels as a value too: `1px` or `1%` for example to allow more flexibility